src/Build.UnitTests/Evaluation/ItemEvaluation_Tests.cs

```diff
@@ -120,6 +120,39 @@ public void RemoveShouldPreserveIntermediaryReferences(string content)
             ObjectModelHelpers.AssertItems(new string[0], itemsForI2);
         }
 
+        [Fact]
+        public void RemoveRespectsItemTransform()
+        {
+            var content = @"
+                            <i Include='a;b;c' />
+
+                            <i Remove='@(i->WithMetadataValue(`Filename`, `b`))' />";
```

Comment:
Curious, why use `Filename` instead of `Identity`?

Reply:
I was copy/pasting from the real repro :) `Identity` does make more sense.